### DISCUSSION TITLE
C# debugger variable value visualisation

### DISCUSSION THREAD

**Discussion by janaka** [OmniSharp/omnisharp-vscode#7807] [Q&A]
Hi, 
Is the variable value visualiser in debug mode, is the code for that in this repo or somewhere else? Can somebody point me in the right direction please?

search terms: visualization visualizer visualiser

## Why I'm interested

When in debug mode and inspecting variable values in the locals or watch window or in the code pain the value visualiser on hover is pretty poor in VS Code. This is especially an issue when inspecting strings with more than a sentence of content (plain text, json, error text etc.). It's enough of a pain that I'm considering making a contribution. 

None of the visualiser extensions I've seen solves the need anywhere near adequately. They open up in a new window. Which is nice sometimes but 90% of the time you want to just hover and inspect full VS use to have a formal extension mechanism for third party visualiser to integrate right in. I'm guessing that doesn't exist - I've searched?

Is there a GH issue already tracking this feature request?

Thanks

**Comment by janaka** [comment]
just found this (american english :/ ) https://github.com/dotnet/vscode-csharp/issues/5023 where somebody is asking the same in 2022. No details.

**Comment by gregg-miskelly** [comment]
TLDR: there isn't a good way to do what you want to do today.

I added more information to #5023 on the right way to do this.

The only option to build a visualizer today is that VS Code does support a contribution point for `debug/variables/context`. I haven't built one, so I can't provide much guidance, but I see that https://github.com/microsoft/vscode-mock-debug has one.
